### PR TITLE
Keylogging (Win32k ETW) API Event (production)

### PR DESCRIPTION
**Pull request by AsuNa-jp**
## Change Summary

Adds new custom api fields for the keylogging events (ETW Win32k API Event provider).

### Sample values

Below are examples of the custom API fields that will be added by this PR. The sample of the document is here ([sample_event.json](https://github.com/elastic/endpoint-package/blob/cc8671e2954da43da6f55946fa78dfa0a02d4b91/package/endpoint/data_stream/api/sample_event.json)) 

**GetAsyncKeyState API's custom API fields**
```
            "message": "Endpoint API event - GetAsyncKeyState",
            "process": {
                "Ext": {
                    "ancestry": [
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTEyMDg0LTE3MDA4MTI5MzguNzg0NzE0MjAw",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTExNTQwLTE3MDA4MTI5MzYuODYxNDY3MDAw",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTY2NC0xNzAwODExNzE1Ljk1MDA2NTIwMA==",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTUyOC0xNzAwODExNzE1LjgyMjg3MTUwMA=="
                    ],
                    "api": {
                        "behaviors": [
                            "rapid_background_polling",
                            "image_indirect_call"
                        ],
                        "metadata": {
                            "background_callcount": 223,
                            "ms_since_last_keyevent": 0
                        },
                        "name": "GetAsyncKeyState"
                    },
```

**SetWindowsHookEx API's custom API fields**
```
            "message": "Endpoint API event - SetWindowsHookEx",
            "process": {
                "Ext": {
                    "ancestry": [
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTM4NTYtMTcwMDgxMTkyMS42ODI1MjYxMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTQ1NjgtMTcwMDgxMTkyMS42NzMxNjcwMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTE3MDQtMTcwMDgxMTY5MC40NzM1MjcyMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTQyMTYtMTcwMDgxMTYxNS45NTM2MzY3MDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTYxMjQtMTcwMDgxMTYxNC44MjM5NzczMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTYwNTYtMTcwMDgxMTYxNC43NjkzODEzMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTY4OC0xNzAwODExNTcxLjc4NjAwNjUwMA==",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTU0OC0xNzAwODExNTcxLjY3NDEyOTcwMA=="
                    ],
                    "api": {
                        "metadata": {
                            "procedure_symbol": "dinput8.dll"
                        },
                        "name": "SetWindowsHookEx",
                        "parameters": {
                            "hook_module": "c:\\windows\\system32\\dinput8.dll",
                            "hook_type": "WH_MOUSE_LL",
                            "procedure": 140707050053760
                        },
                        "summary": "SetWindowsHookEx( WH_MOUSE_LL, dinput8.dll )"
                    },
```

**RegisterRawInputDevices API's custom API fields**
```
            "message": "Endpoint API event - RegisterRawInputDevices",
            "process": {
                "Ext": {
                    "ancestry": [
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTM4NTYtMTcwMDgxMTkyMS42ODI1MjYxMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTQ1NjgtMTcwMDgxMTkyMS42NzMxNjcwMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTE3MDQtMTcwMDgxMTY5MC40NzM1MjcyMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTQyMTYtMTcwMDgxMTYxNS45NTM2MzY3MDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTYxMjQtMTcwMDgxMTYxNC44MjM5NzczMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTYwNTYtMTcwMDgxMTYxNC43NjkzODEzMDA=",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTY4OC0xNzAwODExNTcxLjc4NjAwNjUwMA==",
                        "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTU0OC0xNzAwODExNTcxLjY3NDEyOTcwMA=="
                    ],
                    "api": {
                        "behaviors": [
                            "native_api",
                            "image_indirect_call"
                        ],
                        "metadata": {
                            "return_value": 1,
                            "start_address_allocation_protection": "RCX",
                            "start_address_module": "C:\\Python310\\python.exe",
                            "thread_info_flags": 16,
                            "visible_windows_count": 0,
                            "windows_count": 2
                        },
                        "name": "RegisterRawInputDevices",
                        "parameters": {
                            "flags": "INPUTSINK",
                            "usage": "KEYBOARD",
                            "usage_page": "GENERIC"
                        }
                    },
```

Sample document:

```json


```

## Release Target

 8.12

### For mapping changes:

- [x] I ran `make` after making the schema changes, and committed all changes
- [ ] If these field(s) are "exception"-able, I made a companion PR to Kibana adding it (see [Readme](https://github.com/elastic/endpoint-package#exceptionable))
- [ ] If this is a `metadata` change, I also updated both transform destination schemas to match

### For Transform changes:

- [ ] The new transform successfully starts in Kibana
- [ ] The corresponding transform destination schema was updated if necessary

### CODE REVIEW

File: custom_schemas/custom_api.yml [hunk 1]
```diff
@@ -328,4 +330,116 @@
       type: keyword
       description: >
         The name of the driver object.
-      example: "\\Driver\\npcap"
\ No newline at end of file
+      example: "\\Driver\\npcap"
+
+    - name: parameters.return_value
```
Comment (jdu2600) [inline custom_schemas/custom_api.yml:335]: 🤔
Some of these fields are not strictly function `parameters`.
Perhaps we should put them under `metadata` instead.

**Comment by jdu2600** [comment]
We should also add sample entries for each of the new fields to `package/endpoint/data_stream/api/sample_event.json`
See https://github.com/elastic/endpoint-package/pull/445#issuecomment-1791204268

**Comment by AsuNa-jp** [comment]
@jdu2600 
>We should also add sample entries for each of the new fields to

Thanks for letting me know about above. I was wondering how to add the sample event for my apis, so this was very helpful. Thank you!

**Comment by AsuNa-jp** [comment]
Hi @ashokaditya @tomsonpl 
This PR is now ready to review & merge. When you have a moment could you take a look at this PR?
and, please let me know if I need any modifications or something.

(+ it seem that there is [a bug](https://buildkite.com/elastic/endpoint-package/builds/147#018c0058-a2de-486b-819a-cf0bcc137f5a) in the buildkite side?)
```
  | kibana_version_condition = manifest_doc["conditions"]["kibana.version"]
  | KeyError: 'kibana.version'
  | 🚨 Error: The command exited with status 1
```

**Comment by AsuNa-jp** [comment]
CC: @elastic/security-defend-workflows

**Comment by tomsonpl** [comment]
@pzl Hey Dan, I hope you don't mind that I assigned you here. Could you help with the review? I am not familiar with endpoint-package enough. Thanks!

**Review by ashokaditya — APPROVED** [review]
Looks good to me. The CI steps need to pass though. 

@pzl do you know why https://github.com/elastic/endpoint-package/blob/99d116c85347bcfbf916993d19e3de2cab6cd351/.buildkite/scripts/find_oldest_supported_version.py#L119 doesn't work anymore. Looks like now `manifest_doc["conditions"]["kibana"]["version"]` works instead.

**Review by pzl — APPROVED** [review]
the required CI passes. Buildkite is always broken

this is ok to merge @AsuNa-jp

**Pull request merged**